qemu/tests/qemu_guest_agent.py

```diff
@@ -126,11 +143,62 @@ def _check_ga_service(self, session, cmd_check_status):
         :param cmd_check_status: cmd to check if ga service is started
         '''
-        error_context.context("Check whether qemu-ga service is started.", logging.info)
+        error_context.context("Check whether qemu-ga service is started.",
+                              logging.info)
         s, o = session.cmd_status_output(cmd_check_status)
         return s == 0
 
     @error_context.context_aware
+    def _get_latest_pkg(self):
+        """
+        get latest qemu-guest-agent rpm package url.
+        :return: rpm pkg list
+        """
+        virt_module_stream = self.params.get("virt_module_stream", "")
+        guest_name = self.params.get("guest_name")
+        arch = self.params["vm_arch_name"]
+        download_root = self.params["download_root_url"]
+        query_timeout = 180
+
+        error_context.context("Check if brew command is presented.",
+                              logging.info)
+        if process.system("which brew", ignore_status=True) != 0:
+            self.test.error("Pls install brew command in host.")
+
+        error_context.context("Get latest virt module tag of %s"
+                              " stream." % virt_module_stream,
+                              logging.info)
+        # target release,such as 810,811
+        target_release = re.findall(r'rhel(\d+)-\d+', guest_name, re.I)[0]
+        # get tag pattern,such as module-virt-8.1-80101xxxxx
+        if virt_module_stream == "rhel":
+            # for slow train,didn't know 810 or 811.
+            # module-virt-rhel-801xxx
+            target_release = target_release[:-1]
+        tag_version = "0".join(target_release)
+        # module-virt-8.1-80101 or module-virt-rhel-801
+        platform_tag = "module-virt-%s-%s" % (virt_module_stream,
+                                              tag_version)
+        get_latest_mdl_tag_cmd = "brew list-targets |grep"
+        get_latest_mdl_tag_cmd += " %s |sort -r |head -n 1" % platform_tag
+        latest_mdl_tag = process.system_output(get_latest_mdl_tag_cmd,
+                                               shell=True,
+                                               timeout=query_timeout
+                                               ).strip().split()[0].decode()
+        error_context.context("Get qemu-guest-agent rpm pkg url.",
+                              logging.info)
+        get_brew_latest_pkg_cmd = "brew --quiet --topdir=%s" % download_root
+        get_brew_latest_pkg_cmd += " list-tagged %s" % latest_mdl_tag
+        get_brew_latest_pkg_cmd += " --path --arch=%s" % arch
+        get_brew_latest_pkg_cmd += " |grep qemu-guest-agent-[0-9]"
+
+        rpm_url = process.system_output(get_brew_latest_pkg_cmd,
+                                        shell=True,
+                                        timeout=query_timeout
+                                        ).strip().decode()
+        logging.info("Qemu-guest-agent rpm pkg url is %s" % rpm_url)
+        return rpm_url
+
```

Comment:
@xiagao how to decide slow train or fast train?  e.g sometimes you need to test the latest slow train package or fast train. how to distinguish them?

Reply:
Good question, I set a parameter to decide which virt moduel stream.
    #only for rhel8,virt module stream is rhel or 8.1,8.2,8.3 and so on
    virt_module_stream = rhel
and in internal test, we can use --customsparams like 
--customsparams="virt_module_stream = rhel"

if virt_module_stream = 8.1, will judge it is 8.1.0 or 8.1.1 via guestname

Reply:
@xiagao  As we discussed.  I think guest should install agent packet is the same as host.  e.g host is fast train.  guest should install fast train agent package.   shouldn't install slow package.
If you  can determine qemu-kvm version doesn't affect agent function.  then you may ignore fast or slow train.  only focus on agent package testing.  

In addition, If you install agent package is the same with host.  you don't need to use paramter  "--customsparams="

Reply:
@xiagao  for rhel8 guest in rhel7 host.  only focus on rhel8 agent package testing. 

If you can select test slow or fast train package when testing. I think it is enough.  otherwise, maybe fast train pacakge won't be tested(As guest os install slow train only by default)

Reply:
yes, it does.

Reply:
Ack by: Xiangchun Fu xfu@redhat.com